### PR TITLE
VZ-2047: Add support for delete in VerrazzanoWebLogicWorkload

### PR DESCRIPTION
**Pull request by fredtibbitts**
# Description

PR includes:
- Add support for delete in VerrazzanoWebLogicWorkload
- Additional unit tests to boost coverage to > 80%
- Small refactoring to make some strings constants/shared

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/wlsworkload/weblogicworkload_controller.go [hunk 1]
```diff
@@ -195,33 +211,65 @@ func (r *Reconciler) addLogging(ctx context.Context, log logr.Logger, namespace
...
 	return nil
 }
+
+// handleDelete handles delete processing - we delete the WebLogic domain CR when our VerrazzanoWebLogicWorkload is deleted.
+// returns true if our workload is being deleted, false otherwise
+func (r *Reconciler) handleDelete(ctx context.Context, log logr.Logger, workload *vzapi.VerrazzanoWebLogicWorkload, weblogic *unstructured.Unstructured) (bool, error) {
+	if !workload.ObjectMeta.DeletionTimestamp.IsZero() {
+		if controllers.StringSliceContainsString(workload.ObjectMeta.Finalizers, finalizer) {
+			log.Info("Deleting WebLogic domain CR")
+			if err := r.Delete(ctx, weblogic); err != nil {
```
Comment (kminder) [inline application-operator/controllers/wlsworkload/weblogicworkload_controller.go:251]: If the error is a NotFound you should ignore it and continue to delete the finalizer.

File: application-operator/controllers/wlsworkload/weblogicworkload_controller.go [hunk 2]
```diff
@@ -195,33 +211,65 @@ func (r *Reconciler) addLogging(ctx context.Context, log logr.Logger, namespace
...
 		log.Error(err, "Unable to set logHomeEnabled")
 		return err
 	}
 
 	return nil
 }
+
+// handleDelete handles delete processing - we delete the WebLogic domain CR when our VerrazzanoWebLogicWorkload is deleted.
+// returns true if our workload is being deleted, false otherwise
+func (r *Reconciler) handleDelete(ctx context.Context, log logr.Logger, workload *vzapi.VerrazzanoWebLogicWorkload, weblogic *unstructured.Unstructured) (bool, error) {
```
Comment (kminder) [inline application-operator/controllers/wlsworkload/weblogicworkload_controller.go:247]: I don't think this is necessary if you setup the ownerRef correctly.  That would avoid the need for the finalizer which can be problematic.

File: application-operator/controllers/wlsworkload/weblogicworkload_controller.go (hunk 1 above)
Comment (fredtibbitts) [reply application-operator/controllers/wlsworkload/weblogicworkload_controller.go:251]: Fixed

File: application-operator/controllers/wlsworkload/weblogicworkload_controller.go (hunk 2 above)
Comment (fredtibbitts) [reply application-operator/controllers/wlsworkload/weblogicworkload_controller.go:247]: I will remove all of this and set the owner ref. My fault for not knowing how all of the k8s delete stuff works. I'll need to do the same in the VerrazzanoCoherenceWorkload controller. I'll do both of them in a subsequent PR.

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**